### ISSUE TITLE
fix: device bitcast binary_field_ghash -> uint64 silently returns zeros

### ISSUE THREAD

**Issue by chokobole** [fractalyze/zorch#399]
### Pre-submission checklist

- [x] I have searched the existing issues and this is not a duplicate.

### Affected version / commit

zorch `main` @ `37f8b9e` · jax/jaxlib `0.10.0.dev20260706133158` · zk-dtypes `0.0.8`

### Environment

Linux

### Reproducer

```shell
JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 python - <<'PY'
import jax, jax.numpy as jnp, numpy as np
from jax import lax
jax.config.update("jax_enable_x64", True)

u = jnp.asarray(np.array([[2, 0], [3, 0], [0, 1]], np.uint64))
g = lax.bitcast_convert_type(u, jnp.binary_field_ghash)  # OK: [2, 3, 2**64]
print(lax.bitcast_convert_type(g, jnp.uint64))           # all zeros
PY
```

### Expected behavior

The `binary_field_ghash → uint64` bitcast round-trips the forward direction: `[[2,0],[3,0],[0,1]]` (lo, hi limbs back out).

### Actual behavior

All zeros — **silently**. The forward direction (`uint64[..,2] → binary_field_ghash`) is correct, and the host path is too (`np.asarray(g).tobytes()` equals the source bytes), so only the device reverse-bitcast lowering is broken. A silent wrong result is worse than a crash: any device pipeline reading GHASH limbs back out computes on zeros.

Hit while wiring flock-zorch's `ligero_commit` onto `coding.ReedSolomon(binary_field_ghash)` (fractalyze/flock-zorch#36) — worked around by returning the codeword to uint64 SoA through host bytes. Same binary-field-lowering family as #398.

### Relevant logs / stack trace

```text
u64->ghash: [2 3 18446744073709551616]   (correct)
ghash->u64: [[0 0] [0 0] [0 0]]          (expected [[2 0] [3 0] [0 1]])
```

### Area

compiler

### Relevant knowledge

Verified on the CPU backend (`JAX_PLATFORMS=cpu`); untested on cuda. `zorch.pcs.fold.from_base_field` relies on `lax.bitcast_convert_type(..., dtype)` for extension fields — if binary fields ever route through it, this gap makes that path silently zero as well.

**Comment by chokobole** [comment]
### Root cause = XLA CPU `BitcastDtypesExpander` UB — fixed in fractalyze/xla#241

Same underlying bug as the CPU side of fractalyze/flock-zorch#75. The narrowing path of `xla/hlo/transforms/expanders/bitcast_dtypes_expander.cc` computed the low-bits mask as `(int64_t{1} << output_bit_width) - 1`; for a 128-bit → 64-bit bitcast `output_bit_width == 64`, so `int64_t{1} << 64` is undefined behavior and the mask collapses to `0` on x86. It's visible in the optimized HLO as `u128[] constant("0x0")`, and the following `And` zeroes `binary_field_ghash → uint64`.

**Fix:** fractalyze/xla#241 — well-defined unsigned mask + a `U128 → U64` expander test.

**Verified on CPU** with a jaxlib built from that branch — this issue's exact reproducer now round-trips:
```
u64->ghash: [2, 3, 18446744073709551616]
ghash->u64: [[2, 0], [3, 0], [0, 1]]   (was all zeros)
```
plus batched 1024 round-trip (eager + jit) matches source.

The `ligero_commit` host-bytes workaround can be dropped once a jaxlib with #241 is published and the pin bumps.
